src/test/helper.ts

```diff
@@ -5,38 +5,124 @@
 
 import * as vscode from 'vscode';
 import * as path from 'path';
-
-export let doc: vscode.TextDocument;
-export let editor: vscode.TextEditor;
-export let documentEol: string;
-export let platformEol: string;
+import * as assert from 'assert';
 
 export async function open(docUri: vscode.Uri): Promise<void> {
   try {
-    doc = await vscode.workspace.openTextDocument(docUri);
-    editor = await vscode.window.showTextDocument(doc);
+    const doc = await vscode.workspace.openTextDocument(docUri);
+    await vscode.window.showTextDocument(doc);
   } catch (e) {
     console.error(e);
     throw e;
   }
 }
 
-export function getExtensionId(): string {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const pjson = require('../../package.json');
-  return `${pjson.publisher}.${pjson.name}`;
+export const getDocUri = (p: string): vscode.Uri => {
+  const documentPath = path.resolve(__dirname, '../../test/fixtures', p);
+  return vscode.Uri.file(documentPath);
+};
+
+export async function testCompletion(
+  docUri: vscode.Uri,
+  position: vscode.Position,
+  expectedCompletionList: vscode.CompletionList,
+) {
+  const actualCompletionList = await vscode.commands.executeCommand<vscode.CompletionList>(
+    'vscode.executeCompletionItemProvider',
+    docUri,
+    position,
+  );
+
+  assert.deepStrictEqual(actualCompletionList.items.length, expectedCompletionList.items.length);
+  expectedCompletionList.items.forEach((expectedItem, i) => {
+    const actualItem = actualCompletionList.items[i];
+    assert.deepStrictEqual(actualItem.label, expectedItem.label);
+    assert.deepStrictEqual(actualItem.kind, expectedItem.kind);
+  });
 }
 
-export const testFolderPath = path.resolve(__dirname, '..', '..', 'testFixture');
+export async function testHover(docUri: vscode.Uri, position: vscode.Position, expectedCompletionList: vscode.Hover[]) {
+  const actualhover = (await vscode.commands.executeCommand(
+    'vscode.executeHoverProvider',
+    docUri,
+    position,
+  )) as vscode.Hover[];
```

Comment:
```suggestion
  const actualhover = await vscode.commands.executeCommand<vscode.Hover[]>(
    'vscode.executeHoverProvider',
    docUri,
    position,
  );
```

We can pass the type argument to `executeCommand` here, like you did for the other helpers.

Reply:
Yeah, after the first few it was tedious adjusting the samples and I forgot to circle back. Thanks!